nuget/lib/dependabot/nuget/native_helpers.rb

```diff
@@ -56,27 +57,28 @@ def self.run_nuget_framework_check(project_tfms, package_tfms)
       end
 
       sig do
-        params(repo_root: String, proj_path: String, dependency: Dependency,
-               is_transitive: T::Boolean).returns([String, String])
+        params(repo_root: String, proj_path: String, dependencies: T::Array[Dependabot::Dependency])
+          .returns([String, String])
       end
-      def self.get_nuget_updater_tool_command(repo_root:, proj_path:, dependency:, is_transitive:)
+      def self.get_nuget_updater_tool_command(repo_root:, proj_path:, dependencies:)
         exe_path = File.join(native_helpers_root, "NuGetUpdater", "NuGetUpdater.Cli")
+
         command_parts = [
           exe_path,
           "update",
           "--repo-root",
           repo_root,
           "--solution-or-project",
           proj_path,
-          "--dependency",
-          dependency.name,
-          "--new-version",
-          dependency.version,
-          "--previous-version",
-          dependency.previous_version,
-          is_transitive ? "--transitive" : nil,
           "--verbose"
-        ].compact
+        ]
+
+        dependencies.each do |dep|
+          command_parts << "--dependency"
+          command_parts << <<~DEPENDENCY_JSON
+            {"Name":"#{dep.name}","NewVersion":"#{dep.version}","PreviousVersion":"#{dep.previous_version}","IsTransitive":#{!dep.top_level?}}
```

Comment:
Building the json like this seems brittle. Is there a different approach where we could serialize an object?

Reply:
This can be done with something like (not 100% on the syntax, but it's close):

``` ruby
command_parts << {
  "Name" => dep.name,
  "NewVersion" => dep.version,
  "PreviousVersion" => dep.previous_version,
}.to_json
```

Reply:
Another concern I have here is creating arguments that are too long for the command line. Could we pass the dependencies through a json file. Alternately, the System.CommandLine library we use on the .NET side supports using with RDP files.

Reply:
I can look at this